v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/BigtableChangeStreamsToPubSubIT.java

```diff
@@ -0,0 +1,821 @@
+    } else {
+      Assert.assertNull(valueNode);
+    }
+    Assert.assertEquals(expected.getSourceInstance(), jsonTree.get("sourceInstance").asText());
+    Assert.assertEquals(expected.getSourceCluster(), jsonTree.get("sourceCluster").asText());
+    Assert.assertEquals(expected.getSourceTable(), jsonTree.get("sourceTable").asText());
+  }
+
+  @NotNull
+  private static String generateAppProfileId() {
+    return "cdc_app_profile_" + System.nanoTime();
+  }
+
+  @Before
+  public void setup() throws IOException {
+    pubsubResourceManager =
+        PubsubResourceManager.builder(
+                removeUnsafeCharacters(testName), PROJECT, credentialsProvider)
+            .build();
+    BigtableResourceManager.Builder rmBuilder =
+        BigtableResourceManager.builder(
+            removeUnsafeCharacters(testName), PROJECT, credentialsProvider);
+
+    bigtableResourceManager = rmBuilder.maybeUseStaticInstance().build();
+
+    appProfileId = generateAppProfileId();
+
+    String suffix = "" + System.nanoTime();
+    String topicNameToCreate = "topic-" + suffix;
+    String subscriptionNameToCreate = "subscription-" + suffix;
+    srcTable = "src" + suffix;
+
+    List<BigtableResourceManagerCluster> clusters = new ArrayList<>();
+    clusters.add(BigtableResourceManagerCluster.create(clusterName, TEST_ZONE, 1, StorageType.HDD));
+
+    bigtableResourceManager.createInstance(clusters);
+
+    bigtableResourceManager.createAppProfile(
+        appProfileId, true, Lists.asList(clusterName, new String[]{}));
+
+    BigtableTableSpec cdcTableSpec = new BigtableTableSpec();
+    cdcTableSpec.setCdcEnabled(true);
+    cdcTableSpec.setColumnFamilies(Lists.asList(SOURCE_COLUMN_FAMILY, new String[]{}));
+    bigtableResourceManager.createTable(srcTable, cdcTableSpec);
+
+    topicName = pubsubResourceManager.createTopic(topicNameToCreate);
+    subscriptionName = pubsubResourceManager.createSubscription(topicName,
+        subscriptionNameToCreate);
+  }
+
+  private String removeUnsafeCharacters(String testName) {
+    return testName.replaceAll("[\\[\\]]", "-");
+  }
+
+  @After
+  public void tearDownClass() {
+    ResourceManagerUtils.cleanResources(bigtableResourceManager, pubsubResourceManager);
+  }
+
+  @Override
+  protected Config createConfig(LaunchInfo info) {
+    Config.Builder configBuilder =
+        Config.builder().setJobId(info.jobId()).setProject(PROJECT).setRegion(REGION);
+
+    // For DirectRunner tests, reduce the max time and the interval, as there is no worker required
+    if (System.getProperty("directRunnerTest") != null) {
+      configBuilder =
+          configBuilder
+              .setTimeoutAfter(EXPECTED_REPLICATION_MAX_WAIT_TIME.minus(Duration.ofMinutes(3)))
+              .setCheckAfter(Duration.ofSeconds(5));
+    } else {
+      configBuilder.setTimeoutAfter(EXPECTED_REPLICATION_MAX_WAIT_TIME);
+    }
+
+    return configBuilder.build();
+  }
```

Comment:
You could modify the `@Category` tag on the class to be `@Category({TemplateIntegrationTest.class, SkipDirectRunnerTest.class})` if the DirectRunner tests timeout (and therefore remove this override). I'm not sure we want DirectRunner tests to be taking more than a couple minutes.

@bvolpato thoughts?

Reply:
I think so, DirectRunner tests are supposed to be faster / provide an early signal. Things will be tested on Dataflow before any release regardless, but I think we can ignore direct runner.